src/components/graph-view.js

```diff
@@ -400,10 +394,15 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
       });
 
       // remove node
-      // The timeout avoids a race condition
-      setTimeout(() => {
-        GraphUtils.removeElementFromDom(`node-${nodeId}-container`);
-      });
+      const timeoutId = `nodes-${nodeId}`;
+
+      // cancel an asyncRenderNode animation
+      cancelAnimationFrame(this.nodeTimeouts[timeoutId]);
```

Comment:
This fixes the race condition bug by cancelling the asyncRenderNode method. Now nodes are deleted and stay deleted.